spring-cql/src/test/java/org/springframework/cassandra/test/integration/support/CassandraConnectionProperties.java

```diff
@@ -80,17 +79,47 @@ public long getCqlInitializationTimeout() {
 		return getLong("build.cql.init.timeout");
 	}
 
+	public String getCassandraHost() {
+		return getProperty("build.cassandra.host");
+	}
+
+	public CassandraType getCassandraType() {
+		String property = getProperty("build.cassandra.mode");
+		if (property != null && property.equalsIgnoreCase(CassandraType.EXTERNAL.name())) {
+			return CassandraType.EXTERNAL;
+		}
+		return CassandraType.EMBEDDED;
+	}
+
 	public int getInt(String key) {
 		String property = getProperty(key);
-		return Integer.parseInt(property);
+		try {
+			return Integer.parseInt(property);
+		} catch (NumberFormatException e) {
+			throw new IllegalStateException(resourceName + ": Cannot parse property " + key + " to int: " + property);
+		}
 	}
 
 	public long getLong(String key) {
 		String property = getProperty(key);
-		return Long.parseLong(property);
+		try {
+			return Long.parseLong(property);
+		} catch (NumberFormatException e) {
+			throw new IllegalStateException(resourceName + ": Cannot parse property " + key + " to long: " + property + "");
+		}
 	}
 
 	public boolean getBoolean(String key) {
-		return Boolean.parseBoolean(getProperty(key));
+		String property = getProperty(key);
+		try {
+			return Boolean.parseBoolean(property);
+		} catch (NumberFormatException e) {
```

Comment:
We could even simplify all these typed 'get' property methods to, for example...

```
public int getInt(String propertyName) {
    return convert(propertyName, Integer.class, new Converter<String, Integer.TYPE>() {
        public int convert(String value) {
            return Integer.parseInt(value);
        }
    };
}

protected <T> T convert(String propertyName, Class<T> type, Converter<String, T> converter) {
    String propertyValue = getProperty(propertyName);
    try {
        return converter.convert(propertyValue);
    }
    catch (Exception e) {
        throw IllegalFormatException(String.format("%1$s: cannot parse value [%2$s] of property [%3$s] as a [%4$s]",
            resourceName, propertyValue, propertyName, type.getSimpleName()), e);
    }
}
```

Then, when we move to Java 8, we can simplify the typed 'get' property method to using method references since Spring's `Converter` interface is a functional interface, for example...

```
public int getInt(String propertyName) {
    return convert(propertyName, Integer.class, Integer::parseInt);
}
```

This will make it easier to support additional property types as needed in the future as well.